specification/network/resource-manager/Microsoft.Network/stable/2020-06-01/expressRouteProviderPort.json

```diff
@@ -0,0 +1,208 @@
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "expressRouteProviderPortName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of ExpressRouteProviderPort."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the requested ExpressRouteProviderPort resource.",
+            "schema": {
+              "$ref": "#/definitions/ExpressRouteProviderPort"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ExpressRouteProviderPortGet": {
+            "$ref": "./examples/expressRouteProviderPortGet.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ExpressRouteProviderPortProperties": {
+      "properties": {
+        "portPairDescriptor": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the port pair."
+        },
+        "primaryAzurePort": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the primary port."
+        },
+        "secondaryAzurePort": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the secondary port."
+        },
+        "peeringLocation": {
+          "type": "string",
+          "description": "The peering location of the port pair."
+        },
+        "overprovisionFactor": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Overprovisioning factor for the port pair."
+        },
+        "portBandwidthInMbps": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Bandwidth of the port in Mbps"
+        },
+        "usedBandwidthInMbps": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Used Bandwidth of the port in Mbps"
+        },
+        "remainingBandwidthInMbps": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Remaining Bandwidth of the port in Mbps"
+        }
+      },
+      "description": "Properties of ExpressRouteProviderPort."
+    },
+    "ExpressRouteProviderPort": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ExpressRouteProviderPortProperties",
+          "description": "Properties of the express route Service Provider Port."
+        },
+        "etag": {
```

Comment:
The problem before was that it was inheriting from a definition that had location and tags.  Now it isn't inheriting from anything which is also wrong.  This definition needs a top level "id", "name", and "type" property to conform to standard resource contracts.  I'd recommend inheriting from the "ProxyResource" definition in specification\common-types\resource-management\v1\types.json

Reply:
Updated it.. I was trying to check other examples and found that this is not used in Mircosoft.Network till now. However, looks like  that is the right resource, thanks!